lib/module.js

```diff
@@ -25,6 +25,25 @@ module.exports = function (moduleOptions) {
   if (this.nuxt.options.dev) {
     registerValidator.call(this, options)
   }
+
+  this.nuxt.hook('build:extendRoutes', (routes) => {
+    processRoutes(routes)
+  })
+}
+
+function processRoutes(routes) {
+  for (const route of routes) {
+    route.meta = route.meta || {}
+    route.alias = route.alias || []
+    if (route.path === '/amp' || route.path.indexOf('/amp/') === 0) {
+      route.meta.amp = true
+    } else {
+      route.alias = '/amp' + route.path
```

Comment:
(just asking) Does alias work for child routes? (I mean doesn't it duplicate `/amp` prefix?)

Reply:
You were right, It does duplicate prefix in child routes